### PR TITLE
Fix EZP-26209: Textline legacy search indexing causes TransactionError

### PR DESCRIPTION
**Pull request by blankse**
Alternative to https://github.com/ezsystems/ezpublish-legacy/pull/1268

### CODE REVIEW

**Comment by blankse** [comment]
@andrerom I create the PR without your nitpic.

**Comment by andrerom** [comment]
+1, but @alongosz input on if the max and min is within what most SQL engines support _(given this is on generic interface)_.

File: lib/ezdb/classes/ezdbinterface.php [hunk 1]
```diff
@@ -1809,6 +1809,20 @@ public function countStringSize( $string )
      * @var int One of the eZDB::ERROR_HANDLING_* constants
      */
     protected $errorHandling = eZDB::ERROR_HANDLING_STANDARD;
+
+    /**
+     * Maximal value for int columns
+     *
+     * @var int
+     */
+    public $MAX_INT = 2147483647;
```
Comment (alongosz) [inline lib/ezdb/classes/ezdbinterface.php]: Seems like most reasonable limit supported by every DBMS out-of-the-box.

However these should be constants IMHO // cc @andrerom

**Comment by blankse** [comment]
@alongosz I changed it to constants

**Comment by pkamps** [comment]
It's probably OK to use `const`. But in the unlikely case we would support a DBMS with different values for max/min integers we probably have a hard time to overwrite that value in a child class for that DBMS.
https://stackoverflow.com/questions/13613594/overriding-class-constants-vs-properties

**Comment by alongosz** [comment]
@pkamps I wouldn't recommend overriding it at all as it would cause issues for data migrations between systems.
2^31 is a reasonable value that is supported by most of the DBMS-es if not all of them.

But if you really want to override it later, then property is also bad choice because those values are constant with respect to DBMS, not db handler instance. In this case introducing a method, e.g. `getMaxInt` would make more sense.

**Comment by pkamps** [comment]
+1
I'm convinced, it's very unlikely to have different int max/min values.

**Comment by gggeek** [comment]
If going for a const, I vote for naming it MAX_INT_32BIT, so that it is slightly more clear what is intended purpose and origin are. Fe. it makes it easier to add later MAX_INT_64BIT and have code which uses the latter...

**Pull request merged**